devicetypes/smartthings/zwave-siren.src/zwave-siren.groovy

```diff
@@ -394,6 +403,14 @@ def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cm
 		}
 		result << createEvent([name: "switch", value: isActive ? "on" : "off", displayed: true])
 		result << createEvent([name: "alarm", value: isActive ? "both" : "off", displayed: true])
+	} else if (cmd.notificationType == 0x07) { //Tamper Alert
+		switch (cmd.event) {
+			case 0x00: //Tamper switch is pressed more than 3 sec
+			case 0x03: //Tamper switch is pressed more than 3 sec and released
+				result << createEvent([name: "tamper", value: "detected"])
+				result << createEvent([name: "alarm", value: "both"])
```

Comment:
I don't think you understand my comment. Or maybe I didn't understand the documentation. It is my impression that the 0x03 event is meant to be a tamper clear and the 0x00 event is meant to indicate tamper. Do I have that correct, or is that not your impression?

Reply:
I did it wrong in previous PR. 

I understand this in this way: 
When I hold tamper (0x00) it should be clear/clear tampered status, when I recive 0x03 I should turn on tamper.
If I hang the siren, tamper will be pressed more than 3sec(status clear), tamper should be turned on when somebody(release tamper button) will try to take it.

Reply:
Okay. I'll let you make the changes then.

Reply:
Ok I need you help. 
I made some changes in notificationReport : 
`else if (cmd.notificationType == 0x07) { //Tamper Alert
    	if(cmd.event == 0x00) {
    		log.debug "0x00 ------------------"
                        result << secure(zwave.basicV1.basicSet(value: 0x00))
			result << "delay 3000"
			result << secure(zwave.basicV1.basicGet())
            
                        result += "delay 3000"
                        result += sendEvent(name: "tamper", value: "clear")
        	        result += sendEvent(name: "alarm", value: "off")
         } else if(cmd.event == 0x03) {
    		log.debug "0x03 ------------------"
			result += createEvent([name: "tamper", value: "detected"])
			result += createEvent([name: "alarm", value: "both"])
    	}
	}`

And this part of code 
'                result << secure(zwave.basicV1.basicSet(value: 0x00))
		result << "delay 3000"
		result << secure(zwave.basicV1.basicGet())
           '
is taken from off() and here it doesn't work. When I press button and call off() (from app) device turn off. 

Statuses in app are changing but on siren it doesn't.

Reply:
Well for one you've put `sendEvent` where I think you meant to put `createEvent`. You've also mixed the operators `<<` and `+=`. I don't think that matters, but for style reasons you should be consistent at least in the same function.

I think the thing I'm having trouble with is when the device sends the different alarm notifications. When you "tamper" the device the alarm also turns on? But when you "untamper" the device, the alarm does not turn off, so you're trying to do that via z-wave?

Reply:
Perhaps the device only allows you to clear tamper manually (by pressing a button on the device)?

If so, I might have been mistaken in telling you to handle both notification methods.

Reply:
The alarm turns off but light signals don't. 
About operators I know it was just for my tests. 

When I cleared tamper manually (pressing off()) it was working correctly. I will try with those  notifications.

Reply:
I checked this with operators, with createEvent and there is the same result all the time. I can only turn off light effect when manually call off().

Reply:
Okay. Based on what you've described now, your original way of handling it was correct.

Reply:
Sorry.

Reply:
so I should make a commit where handle only 0x03 event and reset status after manual calling off()?

Reply:
No I think what you had before where you only cleared tamper after you received an "off" from the device.